src/CMakeLists.txt

```diff
@@ -325,8 +325,15 @@ if(NOT TARGET RapidJSON::RapidJSON)
     INTERFACE_INCLUDE_DIRECTORIES "${CMAKE_CURRENT_SOURCE_DIR}/s3select/rapidjson/include")
 endif()
 
-option(WITH_FMT_HEADER_ONLY "use header-only version of fmt library" OFF)
-find_package(fmt 8.1.1...<10.0.0)
+option(WITH_FMT_HEADER_ONLY "use header-only version of fmt library" off)
+option(WITH_SYSTEM_FMT "allow use of system fmt if available" on)
+option(REQUIRE_SYSTEM_FMT "require use of system fmt and fail if not found" off)
+if(WITH_SYSTEM_FMT)
+  find_package(fmt 8.1.1...<10.0.0)
+endif()
+if(REQUIRE_SYSTEM_FMT)
+  find_package(fmt 8.1.1...<10.0.0 REQUIRED)
```

Comment:
nit, would be great if we can avoid repeating the version spec.

Reply:
Doesn't apply given the previous change.